components/admin_console/group_settings/group_details/group_profile_and_settings.jsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
+import AdminPanel from 'components/widgets/admin_console/admin_panel';
+import GroupProfile from 'components/admin_console/group_settings/group_details/group_profile';
+import GroupMention from 'components/admin_console/group_settings/group_details/group_mention';
+
+import {t} from 'utils/i18n';
+
+import LineSwitch from 'components/admin_console/team_channel_settings/line_switch.jsx';
+
+const GroupSettingsToggle = ({isDefault, allowReference, onToggle}) => (
+    <LineSwitch
+        disabled={isDefault}
+        toggled={allowReference}
+        last={allowReference}
+        onToggle={() => {
+            if (isDefault) {
+                return;
+            }
+            onToggle(!allowReference);
+        }}
```

Comment:
[The React docs](https://reactjs.org/docs/handling-events.html) recommend against inline functions because...

> a different callback is created each time [it] renders

... whether this affects performance in this particular case, I don't know.

Reply:
This looks to be the pattern when using the LineSwitch component (see team_modes, channel_modes). I say if we were to change this to make it more performant we should do it across the board (maybe a community ticket?). Will defer the change till then.